botorch/test_functions/base.py

```diff
@@ -37,17 +38,20 @@ def __init__(
                 provided, specifies separate noise standard deviations for each
                 objective in a multiobjective problem.
             negate: If True, negate the function.
+            dtype: The dtype that is used for the bounds of the function.
         """
         super().__init__()
         self.noise_std = noise_std
         self.negate = negate
+        self.dtype = dtype
```

Comment:
why assign this? Seems like the only place we'd need this is to register the `bounds` buffer?

Reply:
You are right, there is no reason for this. Will remove, commit and resolve this comment then.

Reply:
Removed the assignment, hence resolving this comment.